contributed_definitions/nyaml/SPM/NXspm.yaml

```diff
@@ -0,0 +1,832 @@
+                  The time taken by the z-controller to measure physical properties.
+              z_controller_hold(NX_BOOLEAN):
+                exists: optional
+                doc: |
+                  The status of the controller to be held on the previous position, before going to the
+                  next scan point or line to measure the physical properties.
+              record_final_z(NX_BOOLEAN):
+                exists: optional
+                doc: |
+                  The status of the controller to record the final z position after the scan.
+          bias_sweep(NXbias_sweep):
+            doc: |
+              The bais voltage sweep is a common technique used on the substance or sample or environment
+              to study the change in the behavior of the sample or substance or environment due to change
+              in applied  bias voltage.
+            settling_time(NX_NUMBER):
+              unit: NX_TIME
+              doc: |
+                The time is taken to settle the bias voltage at the desired value. On each sweep usually, the system
+                takes time to settle the bias voltage at the next value.
+            scan_region(NXobject):
+              doc: |
+                The scan region (phase space or sub-phase space) is the region where the scan is
+                performed.
+              scan_offset_bias(NX_NUMBER):
+                exists: optional
+                unit: NX_VOLTAGE
+                doc: |
+                  The offset of the bias voltage for bias sweep.
+              scan_range_bias(NX_NUMBER):
+                exists: optional
+                unit: NX_VOLTAGE
+                doc: |
+                  The range of the scan is the length of the bias voltage over which the sweep
+                  scan will be performed.
+              scan_start_bias(NX_NUMBER):
+                unit: NX_VOLTAGE
+                doc: |
+                  The start of the bias scan voltage.
+              scan_end_bias(NX_NUMBER):
+                unit: NX_VOLTAGE
+                doc: |
+                  The end of the bias scan voltage.
+            linear_sweep(NXobject):
+              doc: |
+                The linear sweep is a type of scan where the bias voltage is swept
+                linearly from the starting voltage to the ending voltage.
+              scan_points_bias(NX_NUMBER):
+                doc: |
+                  The number of voltage points the sweep scan to be performed.
+              step_size_bias(NX_NUMBER):
+                unit: NX_VOLTAGE
+                doc: |
+                  The step size of the sweep.
+                  The step size is the difference between the two consecutive bias voltage during the sweep.
+              reset_bias(NX_BOOLEAN):
+                exists: optional
+                doc: |
+                  The reset_bias is used to reset the bias voltage to the starting value after a
+                  sweep is completed.
```

Comment:
In my understanding, bias spectroscopy is specific to STS measurements and should not belong to a general NXspm class, but only to its extension, NXsts.

Reply:
Yeah, sts is a fundamental technique and may come along with stm experiment. Still we have a sts app def that inherits from spm without further modifications. So spm offers to count sts along with stm.
 
Overwrite the docs of NXspm saying the appdef NXspm is also used for NXsts experiment.